### PR TITLE
MULE-17262: Add config to choose remote address request attribute format

### PR DESCRIPTION
**Pull request by balbifm**

### CODE REVIEW

File: src/main/java/org/mule/extension/http/internal/listener/HttpRequestAttributesResolver.java [hunk 1]
```diff
@@ -74,11 +81,22 @@ public HttpRequestAttributes resolve() {
...
-        .remoteAddress(clientConnection.getRemoteHostAddress().getHostString().concat(":")
-            .concat(Integer.toString(clientConnection.getRemoteHostAddress().getPort())))
+        .localAddress(resolveLocalAddress())
+        .remoteAddress(resolveRemoteAddress(clientConnection))
         .clientCertificate(clientConnection::getClientCertificate)
         .build();
   }
+
+  private String resolveLocalAddress() {
+    return remoteAddressFormat.equals(RemoteAddressFormat.HOSTNAME_AND_ADDRESS)
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/http/internal/listener/HttpRequestAttributesResolver.java]: static import

File: src/main/java/org/mule/extension/http/internal/listener/HttpRequestAttributesResolver.java [hunk 2]
```diff
@@ -74,11 +81,22 @@ public HttpRequestAttributes resolve() {
...
+        ? requestContext.getServerConnection().getLocalHostAddress().toString()
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/http/internal/listener/HttpRequestAttributesResolver.java]: This reminds me of the org.mule.extension.http.api.error.HttpError#UNSUPPORTED_MEDIA_TYPE enum. Let me know what you think.

Comment (balbifm) [reply src/main/java/org/mule/extension/http/internal/listener/HttpRequestAttributesResolver.java]: I am not following you

File: src/main/java/org/mule/extension/http/api/listener/server/HttpListenerConfig.java [hunk 3]
```diff
@@ -51,8 +57,18 @@
   @Expression(NOT_SUPPORTED)
   private CorsInterceptorWrapper listenerInterceptors;
 
+  /**
+   * Addresses format used to populate local and remote address property in the HTTP request attributes
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/http/api/listener/server/HttpListenerConfig.java]: properties

File: src/main/java/org/mule/extension/http/api/listener/server/RequestAddressesFormatValueProvider.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import static com.google.common.collect.Sets.newHashSet;
+import static org.mule.runtime.extension.api.values.ValueBuilder.newValue;
+
+import org.mule.runtime.api.value.Value;
+import org.mule.runtime.extension.api.values.ValueBuilder;
+import org.mule.runtime.extension.api.values.ValueProvider;
+
+import java.util.Set;
+
+public class RequestAddressesFormatValueProvider implements ValueProvider {
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/http/api/listener/server/RequestAddressesFormatValueProvider.java]: What's the benefit of using value providers here?

Comment (balbifm) [reply src/main/java/org/mule/extension/http/api/listener/server/RequestAddressesFormatValueProvider.java]: It shows nicer values in the select in Studio

**Pull request merged**